streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamThread.java

```diff
@@ -551,6 +551,7 @@ void runLoop() {
                 if (nextProbingRebalanceMs.get() < time.milliseconds()) {
                     log.info("Triggering the followup rebalance scheduled for {} ms.", nextProbingRebalanceMs.get());
                     mainConsumer.enforceRebalance();
+                    nextProbingRebalanceMs.set(Long.MAX_VALUE);
```

Comment:
This is neither relevant to this PR nor required for correctness, but I noticed the log message above tends to spam the logs in some tests. Since this gets set/reset at the end of every rebalance, we may as well reset it here to avoid an avalanche of `Triggering the followup rebalance...`

Reply:
enforceRebalance is guaranteed not to actually run the assignment logic, right? That will only run during a call to poll, I'm hoping. Otherwise, this line should go before the call.

Reply:
Yep. It just provides a notice to the consumer to enforce that a rebalance will occur on the next poll